enhancements/builds/entitlements.md

```diff
@@ -0,0 +1,446 @@
+ concepts, and protocol
+ - but to try and summarize a key element for us: CSI has two step mounting process. staging and publishing. Staging is 
+ node local and publish is pod local. So we will probably want to stage the content once and publish it for each pod 
+ (which is nothing but creating a bind mount for original volume)
+ - That means we do not have to necessarily read and copy the secret(s) contents on every pod creation
+ - We can build an associated controller that watches the global secret, or per namespace secrets that have been 
+ annotated/labelled, and if possible that somehow engages the CSI plugin to redo the staging step.  This is *somewhat*
+ akin to the injection controller employed by the Global Proxy support.
+ - Moving on from the CSI plugin, an admission controller/webhook path (see open questions regarding "packaging" options there) 
+ that based on the presence of the annotation on the pod (or one of several for per namespace perhaps), and the presence 
+ of a global whitelist that stipulates which namespaces can annotate their pod.  See risks and mitigations for 
+ speculation around the need for more granular control.
+ - can mutate the pod, adding volume mounts of the "entitlement" volume 
+
+###### Build Specific consumption path
+
+Once the "entitlement" volume is mounted in the build pod, the openshift builder image will need to map the pod mounted volumes to the 
+requisite buildah parameters (much like it does today for existing secrets/configmaps mounted into the build pod).
+
+###### Build Specific Secret Injection Option
+
+In lieu of the CSI plugin and k8s admission flow, a build only solution would center around  
+The build controller accessing the global secret and mounting into build pods as it deems fit.
+
+
+##### User Namespace Option
+
+Users could provide secret(s) with a well known annotation in their namespace that the global solution
+noted above can look for and mount in the pod in the same way as the global secret.
+
+The global solution minimally would only look for secrets with such an annotation or label in the namespace of the 
+pod who has the annotation for mounting entitlements.
+
+Labels might be preferable in the per namespace case to facilitate queries. 
+
+This would be a slight step up from having to manually cite those secrets in the BuildConfig Spec after creating them,
+as documented today. 
+
+The per namespace version would act as an override and take precedence over the global copy.
+
+NOTE: a user namespace option quite possibly will be more attractive to a general tekton implementation, as 
+access to the openshift-config namespace is not a lower permission level sort of thing.  In theory, the Build V2
+controller may have similar privileges as the current build V1 controller, but we will have to monitor that 
+situation with respect to features like this.
```

Comment:
it's not the build-controller that needs access to the openshift-config namespace.

this volume mounting/content injection would be happening at a pod level, regardless of why the pod was created (as part of a tekton build, a v1 build, a deployment, etc)

Reply:
forgot to update that note with the CSI revamp - thanks